##### สร้าง function ชื่อ smartDict ซึ่งทำหน้าที่:
    -function รับ input เป็น list 2 ตัวที่มีขนาดเท่ากันโดย listแรกเป็น list ของตัวเลข list ตัวที่สองเป็น list ของ text
    -ถ้าค่าใน list แรกเป็นเลขคู่ ให้ค่าใน dict ที่เป็น output มี key เป้นค่าจากlist ตัวที่ 1 และ value จาก listตัวที่ 2
    -ถ้าค่าใน list แรกเป้นเลขคี่ ให้ค่าใน dict ที่เป็น output มี key เป้นค่าจากlist ตัวที่ 2 และ value จาก listตัวที่ 1
    -output ของ function คือ list ที่สร้างขึ้น

In [219]:
def smartDict(k1,k2):
    f = {}
    for i in range(len(k1)):
        if k1 :
            f[k1[i/2 == 0]] = k2[i]
        else:
            f[k2[i]] = k1[i]
    return(f)

In [261]:
smartDict([1,2],['apple','banana'])

{2: 'apple', 1: 'banana'}

### ข้อ 2 จากไฟล์ Wuhan_nCov_GNome.txt จงเขียนโปรแกรมเพื่อตอบคำถามต่อไปนี้
  -กรดอะมิโนใด (A,T,G,C) มีปริมาณมากที่สุดและมีค่าเท่าใด
  -จงหา sequence ที่มีขนาด 3 และ 4 ที่เกิดขึ้นซ้ำมากที่สุด

In [127]:
import pandas as pd

In [128]:
DNA_df = pd.read_csv('C:\\Users\\User\\Desktop\\Wuhan_nCoV_GNome.txt',header=None)

In [130]:
DNA_df

0
0    ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...
1    CGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTG...
2    TAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGC...
3    TTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCG...
4    CCTGGTTTCAACGAGAAAACACACGTCCAACTCAGTTTGCCTGTTT...
..                                                 ...
423  GTGCAGAATGAATTCTCGTAACTACATAGCACAAGTAGATGTAGTT...
424  TTAATCAGTGTGTAACATTAGGGAGGACTTGAAAGAGCCACCACAT...
425  GATCGAGTGTACAGTGAACAATGCTAGGGAGAGCTGCCTATATGGA...
426  TTTAGTAGTGCTATCCCCATGTGATTTTAATAGCTTCTTAGGAGAA...
427                                      AAAAAAAAAAAAA

[428 rows x 1 columns]

## ข้อ 2.1

In [129]:
DNAlong = 0
for i in range(len(DNA_df)):
    DNAlong = DNAlong+len(DNA_df.iloc[i][0])

In [131]:
print(f'sequence is {DNAlong} characters long')

sequence is 29903 characters long


In [64]:
def char1(seq,char):
    charcount = 0
    for i in seq :
        if i == char:
            charcount += 1
    return charcount

In [65]:
Along = 0
Tlong = 0
Glong = 0
Clong = 0
for i in range(len(DNA_df)):
    Along += char1(DNA_df.iloc[i][0],'A')
    Tlong += char1(DNA_df.iloc[i][0],'T')
    Glong += char1(DNA_df.iloc[i][0],'G')
    Clong += char1(DNA_df.iloc[i][0],'C')
print(f'A = {Along} T = {Tlong} G ={Glong} C = {Clong}')

A = 8954 T = 9594 G =5863 C = 5492


### ตอบ กรดอะมิโนที่มีค่ามากที่สุด คือ T มี 9594 ตัว

## ข้อ 2.2

## ตัวที่ซ้ำมากสุด 4 ตัว

In [94]:
listp = 'ATTTATAGCGC'

In [95]:
def str1(seq):
    str23 = []
    for i in range(len(seq)-3):
        str23.append(seq[i:i+4])
    return str23

In [96]:
str1(listp)

['ATTT', 'TTTA', 'TTAT', 'TATA', 'ATAG', 'TAGC', 'AGCG', 'GCGC']

In [97]:
liststr4 = []
for i in range(len(DNA_df)):
    liststr4 += str1(DNA_df.iloc[i][0])

In [98]:
len(liststr4)

28619

In [99]:
unique_4char=set(liststr4)

In [100]:
unique_4char

{'AAAA',
 'AAAC',
 'AAAG',
 'AAAT',
 'AACA',
 'AACC',
 'AACG',
 'AACT',
 'AAGA',
 'AAGC',
 'AAGG',
 'AAGT',
 'AATA',
 'AATC',
 'AATG',
 'AATT',
 'ACAA',
 'ACAC',
 'ACAG',
 'ACAT',
 'ACCA',
 'ACCC',
 'ACCG',
 'ACCT',
 'ACGA',
 'ACGC',
 'ACGG',
 'ACGT',
 'ACTA',
 'ACTC',
 'ACTG',
 'ACTT',
 'AGAA',
 'AGAC',
 'AGAG',
 'AGAT',
 'AGCA',
 'AGCC',
 'AGCG',
 'AGCT',
 'AGGA',
 'AGGC',
 'AGGG',
 'AGGT',
 'AGTA',
 'AGTC',
 'AGTG',
 'AGTT',
 'ATAA',
 'ATAC',
 'ATAG',
 'ATAT',
 'ATCA',
 'ATCC',
 'ATCG',
 'ATCT',
 'ATGA',
 'ATGC',
 'ATGG',
 'ATGT',
 'ATTA',
 'ATTC',
 'ATTG',
 'ATTT',
 'CAAA',
 'CAAC',
 'CAAG',
 'CAAT',
 'CACA',
 'CACC',
 'CACG',
 'CACT',
 'CAGA',
 'CAGC',
 'CAGG',
 'CAGT',
 'CATA',
 'CATC',
 'CATG',
 'CATT',
 'CCAA',
 'CCAC',
 'CCAG',
 'CCAT',
 'CCCA',
 'CCCC',
 'CCCG',
 'CCCT',
 'CCGA',
 'CCGC',
 'CCGG',
 'CCGT',
 'CCTA',
 'CCTC',
 'CCTG',
 'CCTT',
 'CGAA',
 'CGAC',
 'CGAG',
 'CGAT',
 'CGCA',
 'CGCC',
 'CGCG',
 'CGCT',
 'CGGA',
 'CGGC',
 'CGGG',
 'CGGT',
 'CGTA',
 'CGTC',
 'CGTG',
 

In [101]:
len(unique_4char)

256

In [102]:
dict_4char = {}
for i in unique_4char:
    num = 0
    for j in liststr4:
        if i == j:
            num += 1
    dict_4char[i] = num

In [132]:
MaxV = -100
Maxkey = None
for i in dict_4char.keys():
    if dict_4char[i] > MaxV:
        MaxV = dict_4char[i]
        Maxkey = i

In [133]:
print(f'Max 4-char-long sequence is {Maxkey} with frequency equals {MaxV}')

Max 4-char-long sequence is TGTT with frequency equals 318


### ตอบ เกิดขึ้นซ้ำกันมากที่สุด คือ TGTT 318

## ตัวที่ซ้ำ 3 ตัว

In [105]:
listp = 'ATTTATAGCGC'

In [106]:
def str1(seq):
    str23 = []
    for i in range(len(seq)-2):
        str23.append(seq[i:i+3])
    return str23

In [107]:
str1(listp)

['ATT', 'TTT', 'TTA', 'TAT', 'ATA', 'TAG', 'AGC', 'GCG', 'CGC']

In [108]:
liststr3 = []
for i in range(len(DNA_df)):
    liststr3 += str1(DNA_df.iloc[i][0])

In [110]:
len(liststr3)

29047

In [114]:
unique_3char=set(liststr3)

In [115]:
unique_3char

{'AAA',
 'AAC',
 'AAG',
 'AAT',
 'ACA',
 'ACC',
 'ACG',
 'ACT',
 'AGA',
 'AGC',
 'AGG',
 'AGT',
 'ATA',
 'ATC',
 'ATG',
 'ATT',
 'CAA',
 'CAC',
 'CAG',
 'CAT',
 'CCA',
 'CCC',
 'CCG',
 'CCT',
 'CGA',
 'CGC',
 'CGG',
 'CGT',
 'CTA',
 'CTC',
 'CTG',
 'CTT',
 'GAA',
 'GAC',
 'GAG',
 'GAT',
 'GCA',
 'GCC',
 'GCG',
 'GCT',
 'GGA',
 'GGC',
 'GGG',
 'GGT',
 'GTA',
 'GTC',
 'GTG',
 'GTT',
 'TAA',
 'TAC',
 'TAG',
 'TAT',
 'TCA',
 'TCC',
 'TCG',
 'TCT',
 'TGA',
 'TGC',
 'TGG',
 'TGT',
 'TTA',
 'TTC',
 'TTG',
 'TTT'}

In [120]:
dict_3char = {}
for i in unique_3char:
    num = 0
    for j in liststr3:
        if i == j:
            num += 1
    dict_3char[i] = num

In [121]:
MaxV = -100
Maxkey = None
for i in dict_3char.keys():
    if dict_3char[i] > MaxV:
        MaxV = dict_3char[i]
        Maxkey = i

In [122]:
print(f'Max 3-char-long sequence is {Maxkey} with frequency equals {MaxV}')

Max 3-char-long sequence is TTT with frequency equals 984


### ตอบ เกิดขึ้นซ้ำกันมากที่สุด คือ TTT 984

# ข้อ 3

## 3.1  
จากไฟล์ data.xlsx ใหม่ที่มี 2 columns โดย columns ชื่อว่า ภาษาอิสาน และมีค่าใน columm เป็นค่าจาก columm ภาษาอีสานเดิมแต่เปลี่ยน ค่าเป็น '/' เป็น ',' และ colummที่สองชื่อว่า ภาษาไทย และมีค่าใน columm เป้นค่าจาก columm ภาษาไทยเดิมแต่เปลี่ยน ค่าที่เป็น '/' เป็น ','

In [140]:
import pandas as pd

In [141]:
dt = pd.read_excel('dataL.xlsx')

In [142]:
dt

No                                          ภาษาอีสาน  \
0    1.0                          สวัสดี/สุผู้สุคน/เด้อ/จ้า   
1    2.0       กิน/ข้าว/ก่อน/เด้อ/จ้า/ ไผ/สิ/ถ่า/เค้า/ดาวน์   
2    3.0  เค้า/ดาว/กะ/ขอ/ให้/เค้า/ดาว/อย่าง/มี/ความ/สุข/...   
3    4.0  นี่/จ้า/อา/หาร/เย็น/เป็น/ลา/ซาน/ย่า/กับ/ผัก/สด...   
4    5.0                     ขอ/ให้/เชิญ/แซ่บ/เด้อ/พี่/น้อง   
5    6.0         แมท/ธิว/บอก/ว่า/อี/แม่/อย่า/พึ่ง/หลับ/เด้อ   
6    7.0                                       เงียบ/แฮงจัด   
7    8.0          บ่/มี/ความ/รื่น/เริง/อี/หยัง/เลย/ข้าง/นอก   
8    9.0               มื้อ/นี้/สิ/เบิ่ง/พุ/ตอน/หก/ทุ่ม/จ่ะ   
9   10.0             ถ้า/ฝน/ตก/หิ/มะ/ตก/คึ/สิ/บ่/ได้/ออก/ไป   
10  11.0  ปี/ที่/แล้ว/ออก/ไป/เบิ่ง/เขา/จุด/พุ/มัน/เป็น/ฝ...   
11  12.0                        เพิ่น/ก้ม/แบบ/คน/เมา/นะ/จ้ะ   
12  13.0  ก้ม/ก้ม/ส่อง/พุ/เพิ่น/จุด/ละ/เพิ่น/กะ/ก้ม/ส่อง...   
13  14.0            เพิ่น/สิ/โซ/กล้าม/เพิ่น/หั่น/นะ/พี่น้อง   
14  15.0  โอ/เค/จ้า/สุผู้สุคน/มื้อ/นี้/จั่ง/ใด๋/กะ/ขอบ/ค...   
15  16.0  จ้า/พ้อ/กัน/ใหม่/ใน/คลิป/วิ/ดี/โอ/หน้า/สวัสดี/...   
16   NaN                                                NaN   

                                              ภาษาไทย     เริ่ม        จบ  \
0                               สวัสดี/ทุก ๆ คน/นะ/คะ  00:00:16  00:00:18   
1           กิน/ข้าว/ก่อน/นะ/จ้ะ/ใคร/จะ/รอ/เค้า/ดาวน์  00:02:14  00:02:18   
2   เค้า/ดาวน์/ก็/ขอ/ให้/เค้า/ดาน์/อย่าง/มี/ความ/ส...  00:02:21  00:02:25   
3   นี่/จ้า/อา/หาร/เย็น/เป็น/ลา/ซาน/ย่า/กับ/ผัก/สด...  00:02:33  00:02:37   
4                         ขอ/ให้/เชิญ/อร่อย/นะ/ทุก/คน  00:02:38  00:02:41   
5           แมท/ธิว/บอก/ว่า/คุณ/แม่/อย่า/พึ่ง/หลับ/นะ  00:03:08  00:03:08   
6                                         เงียบ/มาก ๆ  00:09:51  00:09:53   
7            ไม่/มี/ความ/รื่น/เริง/อะ/ไร/เลย/ด้าน/นอก  00:10:07  00:10:11   
8                  วัน/นี้/จะ/ดู/พุ/ตอน/หก/ทุ่ม/นะ/คะ  00:10:29  00:10:33   
9             ถ้า/ฝน/ตก/หิ/มะ/ตก/คง/จะ/ไม่/ได้/ออก/ไป  00:10:40  00:10:43   
10  ปี/ที่/แล้ว/ออก/ไป/ดู/เขา/จุด/พุ/มัน/เป็น/ฝน/ห...  00:10:52  00:10:56   
11                         เขา/ก้ม/แบบ/คน/เมา/อ่ะ/ค่ะ  00:11:09  00:11:12   
12  ก้ม/ก้ม/ส่อง/พุ/เขา/จุด/ละ/เขา/ก็/ก้ม/ส่อง/เขา...  00:11:14  00:11:18   
13              เขา/จะ/โชว์/กล้าม/เขา/นั่น/แหละ/ทุกคน  00:13:11  00:13:13   
14  โอ/เค/จ้า/ทุก ๆ  คน/วัน/นี้/ยัง/ไง/ก็/ขอบ/คุณ/...  00:15:23  00:15:26   
15  จ้า/เจอ/กัน/ใหม่/ใน/คลิป/วิ/ดี/โอ/หน้า/สวัสดี/...  00:15:30  00:15:33   
16                                                NaN       NaN       NaN   

    จบ-เริ่ม                                                G2P  
0   00:00:02  s-a-z^-1 w-a-t^-1 d-ii-z^-0 *s-u-z^-1 *ph-uu-z...  
1   00:00:04  k-i-n^-0 *kh-aa-w^-2 *k-@@-n^-1 *d-qq-z^-2 *c-...  
2   00:00:04  kh-a-w^-3 *d-aa-w^-0 *k-a-z^-1 *kh-@@-z^-4 *h-...  
3   00:00:04  n-ii-z^-2 *c-aa-z^-2 *z-aa-z^-0 *h-aa-n^-4 *j-...  
4   00:00:03  kh-@@-z^-4 *h-a-j^-2 *ch-qq-n^-0 *s-xx-p^-2 *d...  
5   00:00:00  m-xx-t^-1 *th-i-w^-0 *b-@@-k^-1 *w-aa-z^-2 *z-...  
6   00:00:02                  ng-iia-p^-2 *h-xx-ng^-0 *c-a-t^-1  
7   00:00:04  b-@@-z^-1 *m-ii-z^-0 *khw-aa-m^-0 *r-vv-n^-2 *...  
8   00:00:04  m-vv-z^-3 *n-ii-z^-3 *s-i-z^-1 *b-q-ng^-1 *ph-...  
9   00:00:03  th-aa-z^-2 *f-o-n^-4 *t-o-k^-1 *h-i-z^-1 *m-a-...  
10  00:00:04  p-ii-z^-0 *th-ii-z^-2 *l-xx-w^-3 *z-@@-k^-1 *p...  
11  00:00:03  ph-qq-n^-2 *k-o-m^-3 *b-xx-p^-1 *kh-o-n^-0 *m-...  
12  00:00:04  k-o-m^-3 *k-o-m^-3 *s-@-ng^-1 *ph-u-z^-3 *ph-q...  
13  00:00:02  ph-qq-n^-2 *s-i-z^-1 *s-oo-z^-0 *kl-aa-m^-2 *p...  
14  00:00:03  z-oo-z^-0 *kh-ee-z^-0 *c-aa-z^-2\n*s-u-z^-1 *p...  
15  00:00:03  c-aa-z^-2 *ph-@@-z^-3 *k-a-n^-0 *m-a-j^-1 *n-a...  
16  00:00:49                                                NaN

In [143]:
dt = dt.dropna()

In [153]:
dt.iloc[0,1].replace("/", ",")

'สวัสดี,สุผู้สุคน,เด้อ,จ้า'

In [154]:
dt

No                                          ภาษาอีสาน  \
0    1.0                          สวัสดี/สุผู้สุคน/เด้อ/จ้า   
1    2.0       กิน/ข้าว/ก่อน/เด้อ/จ้า/ ไผ/สิ/ถ่า/เค้า/ดาวน์   
2    3.0  เค้า/ดาว/กะ/ขอ/ให้/เค้า/ดาว/อย่าง/มี/ความ/สุข/...   
3    4.0  นี่/จ้า/อา/หาร/เย็น/เป็น/ลา/ซาน/ย่า/กับ/ผัก/สด...   
4    5.0                     ขอ/ให้/เชิญ/แซ่บ/เด้อ/พี่/น้อง   
5    6.0         แมท/ธิว/บอก/ว่า/อี/แม่/อย่า/พึ่ง/หลับ/เด้อ   
6    7.0                                       เงียบ/แฮงจัด   
7    8.0          บ่/มี/ความ/รื่น/เริง/อี/หยัง/เลย/ข้าง/นอก   
8    9.0               มื้อ/นี้/สิ/เบิ่ง/พุ/ตอน/หก/ทุ่ม/จ่ะ   
9   10.0             ถ้า/ฝน/ตก/หิ/มะ/ตก/คึ/สิ/บ่/ได้/ออก/ไป   
10  11.0  ปี/ที่/แล้ว/ออก/ไป/เบิ่ง/เขา/จุด/พุ/มัน/เป็น/ฝ...   
11  12.0                        เพิ่น/ก้ม/แบบ/คน/เมา/นะ/จ้ะ   
12  13.0  ก้ม/ก้ม/ส่อง/พุ/เพิ่น/จุด/ละ/เพิ่น/กะ/ก้ม/ส่อง...   
13  14.0            เพิ่น/สิ/โซ/กล้าม/เพิ่น/หั่น/นะ/พี่น้อง   
14  15.0  โอ/เค/จ้า/สุผู้สุคน/มื้อ/นี้/จั่ง/ใด๋/กะ/ขอบ/ค...   
15  16.0  จ้า/พ้อ/กัน/ใหม่/ใน/คลิป/วิ/ดี/โอ/หน้า/สวัสดี/...   

                                              ภาษาไทย     เริ่ม        จบ  \
0                               สวัสดี/ทุก ๆ คน/นะ/คะ  00:00:16  00:00:18   
1           กิน/ข้าว/ก่อน/นะ/จ้ะ/ใคร/จะ/รอ/เค้า/ดาวน์  00:02:14  00:02:18   
2   เค้า/ดาวน์/ก็/ขอ/ให้/เค้า/ดาน์/อย่าง/มี/ความ/ส...  00:02:21  00:02:25   
3   นี่/จ้า/อา/หาร/เย็น/เป็น/ลา/ซาน/ย่า/กับ/ผัก/สด...  00:02:33  00:02:37   
4                         ขอ/ให้/เชิญ/อร่อย/นะ/ทุก/คน  00:02:38  00:02:41   
5           แมท/ธิว/บอก/ว่า/คุณ/แม่/อย่า/พึ่ง/หลับ/นะ  00:03:08  00:03:08   
6                                         เงียบ/มาก ๆ  00:09:51  00:09:53   
7            ไม่/มี/ความ/รื่น/เริง/อะ/ไร/เลย/ด้าน/นอก  00:10:07  00:10:11   
8                  วัน/นี้/จะ/ดู/พุ/ตอน/หก/ทุ่ม/นะ/คะ  00:10:29  00:10:33   
9             ถ้า/ฝน/ตก/หิ/มะ/ตก/คง/จะ/ไม่/ได้/ออก/ไป  00:10:40  00:10:43   
10  ปี/ที่/แล้ว/ออก/ไป/ดู/เขา/จุด/พุ/มัน/เป็น/ฝน/ห...  00:10:52  00:10:56   
11                         เขา/ก้ม/แบบ/คน/เมา/อ่ะ/ค่ะ  00:11:09  00:11:12   
12  ก้ม/ก้ม/ส่อง/พุ/เขา/จุด/ละ/เขา/ก็/ก้ม/ส่อง/เขา...  00:11:14  00:11:18   
13              เขา/จะ/โชว์/กล้าม/เขา/นั่น/แหละ/ทุกคน  00:13:11  00:13:13   
14  โอ/เค/จ้า/ทุก ๆ  คน/วัน/นี้/ยัง/ไง/ก็/ขอบ/คุณ/...  00:15:23  00:15:26   
15  จ้า/เจอ/กัน/ใหม่/ใน/คลิป/วิ/ดี/โอ/หน้า/สวัสดี/...  00:15:30  00:15:33   

    จบ-เริ่ม                                                G2P  
0   00:00:02  s-a-z^-1 w-a-t^-1 d-ii-z^-0 *s-u-z^-1 *ph-uu-z...  
1   00:00:04  k-i-n^-0 *kh-aa-w^-2 *k-@@-n^-1 *d-qq-z^-2 *c-...  
2   00:00:04  kh-a-w^-3 *d-aa-w^-0 *k-a-z^-1 *kh-@@-z^-4 *h-...  
3   00:00:04  n-ii-z^-2 *c-aa-z^-2 *z-aa-z^-0 *h-aa-n^-4 *j-...  
4   00:00:03  kh-@@-z^-4 *h-a-j^-2 *ch-qq-n^-0 *s-xx-p^-2 *d...  
5   00:00:00  m-xx-t^-1 *th-i-w^-0 *b-@@-k^-1 *w-aa-z^-2 *z-...  
6   00:00:02                  ng-iia-p^-2 *h-xx-ng^-0 *c-a-t^-1  
7   00:00:04  b-@@-z^-1 *m-ii-z^-0 *khw-aa-m^-0 *r-vv-n^-2 *...  
8   00:00:04  m-vv-z^-3 *n-ii-z^-3 *s-i-z^-1 *b-q-ng^-1 *ph-...  
9   00:00:03  th-aa-z^-2 *f-o-n^-4 *t-o-k^-1 *h-i-z^-1 *m-a-...  
10  00:00:04  p-ii-z^-0 *th-ii-z^-2 *l-xx-w^-3 *z-@@-k^-1 *p...  
11  00:00:03  ph-qq-n^-2 *k-o-m^-3 *b-xx-p^-1 *kh-o-n^-0 *m-...  
12  00:00:04  k-o-m^-3 *k-o-m^-3 *s-@-ng^-1 *ph-u-z^-3 *ph-q...  
13  00:00:02  ph-qq-n^-2 *s-i-z^-1 *s-oo-z^-0 *kl-aa-m^-2 *p...  
14  00:00:03  z-oo-z^-0 *kh-ee-z^-0 *c-aa-z^-2\n*s-u-z^-1 *p...  
15  00:00:03  c-aa-z^-2 *ph-@@-z^-3 *k-a-n^-0 *m-a-j^-1 *n-a...

In [155]:
def replacestr(col1):
    co1 = col1.replace("/", ",")
    return co1

In [156]:
replacestr(dt.iloc[0,1])

'สวัสดี,สุผู้สุคน,เด้อ,จ้า'

In [157]:
list_1 = []
list_2 = []
for i in range(len(dt)):
    list_1.append(replacestr(dt.iloc[i,1]))
    list_2.append(replacestr(dt.iloc[i,2]))

In [160]:
dt['ภาษาอีสานใหม่'] = list_1
dt['ภาษาไทยใหม่'] = list_2

C:\Users\User\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
C:\Users\User\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [161]:
dt

No                                          ภาษาอีสาน  \
0    1.0                          สวัสดี/สุผู้สุคน/เด้อ/จ้า   
1    2.0       กิน/ข้าว/ก่อน/เด้อ/จ้า/ ไผ/สิ/ถ่า/เค้า/ดาวน์   
2    3.0  เค้า/ดาว/กะ/ขอ/ให้/เค้า/ดาว/อย่าง/มี/ความ/สุข/...   
3    4.0  นี่/จ้า/อา/หาร/เย็น/เป็น/ลา/ซาน/ย่า/กับ/ผัก/สด...   
4    5.0                     ขอ/ให้/เชิญ/แซ่บ/เด้อ/พี่/น้อง   
5    6.0         แมท/ธิว/บอก/ว่า/อี/แม่/อย่า/พึ่ง/หลับ/เด้อ   
6    7.0                                       เงียบ/แฮงจัด   
7    8.0          บ่/มี/ความ/รื่น/เริง/อี/หยัง/เลย/ข้าง/นอก   
8    9.0               มื้อ/นี้/สิ/เบิ่ง/พุ/ตอน/หก/ทุ่ม/จ่ะ   
9   10.0             ถ้า/ฝน/ตก/หิ/มะ/ตก/คึ/สิ/บ่/ได้/ออก/ไป   
10  11.0  ปี/ที่/แล้ว/ออก/ไป/เบิ่ง/เขา/จุด/พุ/มัน/เป็น/ฝ...   
11  12.0                        เพิ่น/ก้ม/แบบ/คน/เมา/นะ/จ้ะ   
12  13.0  ก้ม/ก้ม/ส่อง/พุ/เพิ่น/จุด/ละ/เพิ่น/กะ/ก้ม/ส่อง...   
13  14.0            เพิ่น/สิ/โซ/กล้าม/เพิ่น/หั่น/นะ/พี่น้อง   
14  15.0  โอ/เค/จ้า/สุผู้สุคน/มื้อ/นี้/จั่ง/ใด๋/กะ/ขอบ/ค...   
15  16.0  จ้า/พ้อ/กัน/ใหม่/ใน/คลิป/วิ/ดี/โอ/หน้า/สวัสดี/...   

                                              ภาษาไทย     เริ่ม        จบ  \
0                               สวัสดี/ทุก ๆ คน/นะ/คะ  00:00:16  00:00:18   
1           กิน/ข้าว/ก่อน/นะ/จ้ะ/ใคร/จะ/รอ/เค้า/ดาวน์  00:02:14  00:02:18   
2   เค้า/ดาวน์/ก็/ขอ/ให้/เค้า/ดาน์/อย่าง/มี/ความ/ส...  00:02:21  00:02:25   
3   นี่/จ้า/อา/หาร/เย็น/เป็น/ลา/ซาน/ย่า/กับ/ผัก/สด...  00:02:33  00:02:37   
4                         ขอ/ให้/เชิญ/อร่อย/นะ/ทุก/คน  00:02:38  00:02:41   
5           แมท/ธิว/บอก/ว่า/คุณ/แม่/อย่า/พึ่ง/หลับ/นะ  00:03:08  00:03:08   
6                                         เงียบ/มาก ๆ  00:09:51  00:09:53   
7            ไม่/มี/ความ/รื่น/เริง/อะ/ไร/เลย/ด้าน/นอก  00:10:07  00:10:11   
8                  วัน/นี้/จะ/ดู/พุ/ตอน/หก/ทุ่ม/นะ/คะ  00:10:29  00:10:33   
9             ถ้า/ฝน/ตก/หิ/มะ/ตก/คง/จะ/ไม่/ได้/ออก/ไป  00:10:40  00:10:43   
10  ปี/ที่/แล้ว/ออก/ไป/ดู/เขา/จุด/พุ/มัน/เป็น/ฝน/ห...  00:10:52  00:10:56   
11                         เขา/ก้ม/แบบ/คน/เมา/อ่ะ/ค่ะ  00:11:09  00:11:12   
12  ก้ม/ก้ม/ส่อง/พุ/เขา/จุด/ละ/เขา/ก็/ก้ม/ส่อง/เขา...  00:11:14  00:11:18   
13              เขา/จะ/โชว์/กล้าม/เขา/นั่น/แหละ/ทุกคน  00:13:11  00:13:13   
14  โอ/เค/จ้า/ทุก ๆ  คน/วัน/นี้/ยัง/ไง/ก็/ขอบ/คุณ/...  00:15:23  00:15:26   
15  จ้า/เจอ/กัน/ใหม่/ใน/คลิป/วิ/ดี/โอ/หน้า/สวัสดี/...  00:15:30  00:15:33   

    จบ-เริ่ม                                                G2P  \
0   00:00:02  s-a-z^-1 w-a-t^-1 d-ii-z^-0 *s-u-z^-1 *ph-uu-z...   
1   00:00:04  k-i-n^-0 *kh-aa-w^-2 *k-@@-n^-1 *d-qq-z^-2 *c-...   
2   00:00:04  kh-a-w^-3 *d-aa-w^-0 *k-a-z^-1 *kh-@@-z^-4 *h-...   
3   00:00:04  n-ii-z^-2 *c-aa-z^-2 *z-aa-z^-0 *h-aa-n^-4 *j-...   
4   00:00:03  kh-@@-z^-4 *h-a-j^-2 *ch-qq-n^-0 *s-xx-p^-2 *d...   
5   00:00:00  m-xx-t^-1 *th-i-w^-0 *b-@@-k^-1 *w-aa-z^-2 *z-...   
6   00:00:02                  ng-iia-p^-2 *h-xx-ng^-0 *c-a-t^-1   
7   00:00:04  b-@@-z^-1 *m-ii-z^-0 *khw-aa-m^-0 *r-vv-n^-2 *...   
8   00:00:04  m-vv-z^-3 *n-ii-z^-3 *s-i-z^-1 *b-q-ng^-1 *ph-...   
9   00:00:03  th-aa-z^-2 *f-o-n^-4 *t-o-k^-1 *h-i-z^-1 *m-a-...   
10  00:00:04  p-ii-z^-0 *th-ii-z^-2 *l-xx-w^-3 *z-@@-k^-1 *p...   
11  00:00:03  ph-qq-n^-2 *k-o-m^-3 *b-xx-p^-1 *kh-o-n^-0 *m-...   
12  00:00:04  k-o-m^-3 *k-o-m^-3 *s-@-ng^-1 *ph-u-z^-3 *ph-q...   
13  00:00:02  ph-qq-n^-2 *s-i-z^-1 *s-oo-z^-0 *kl-aa-m^-2 *p...   
14  00:00:03  z-oo-z^-0 *kh-ee-z^-0 *c-aa-z^-2\n*s-u-z^-1 *p...   
15  00:00:03  c-aa-z^-2 *ph-@@-z^-3 *k-a-n^-0 *m-a-j^-1 *n-a...   

                                        ภาษาอีสานใหม่  \
0                           สวัสดี,สุผู้สุคน,เด้อ,จ้า   
1        กิน,ข้าว,ก่อน,เด้อ,จ้า, ไผ,สิ,ถ่า,เค้า,ดาวน์   
2   เค้า,ดาว,กะ,ขอ,ให้,เค้า,ดาว,อย่าง,มี,ความ,สุข,...   
3   นี่,จ้า,อา,หาร,เย็น,เป็น,ลา,ซาน,ย่า,กับ,ผัก,สด...   
4                      ขอ,ให้,เชิญ,แซ่บ,เด้อ,พี่,น้อง   
5          แมท,ธิว,บอก,ว่า,อี,แม่,อย่า,พึ่ง,หลับ,เด้อ   
6                                        เงียบ,แฮงจัด   
7           บ่,มี,ความ,รื่น,

## 3.2
สร้างตาราง Dictionary โดยตารางประกอบด้วยสอง columm คือ ภาษาอีสาน และภาษาไทย โดยค่าใน columm ภาษาอีสาน คืออำในภาษาอีสานและ columm ภาษาไทย คือคำแปลเป็นภาษากลางของภาษาอีสานใน columm แรก (ใช้ค่าจากตาราง data.xlsx โดยคำที่อยุ่ในลำดับเดียวกันของ columm ภาษาอีสานและภาษาไทย คือคำแปลขแงกันและกันถ้า จำนวนค่าของทั้งสอง columm เท่ากัน -ถ้าจำนวนคำระหว่างสอง columm ไม่เท่ากันให้ข้าม record ไป)

In [188]:
dt

No                                          ภาษาอีสาน  \
0    1.0                          สวัสดี/สุผู้สุคน/เด้อ/จ้า   
1    2.0       กิน/ข้าว/ก่อน/เด้อ/จ้า/ ไผ/สิ/ถ่า/เค้า/ดาวน์   
2    3.0  เค้า/ดาว/กะ/ขอ/ให้/เค้า/ดาว/อย่าง/มี/ความ/สุข/...   
3    4.0  นี่/จ้า/อา/หาร/เย็น/เป็น/ลา/ซาน/ย่า/กับ/ผัก/สด...   
4    5.0                     ขอ/ให้/เชิญ/แซ่บ/เด้อ/พี่/น้อง   
5    6.0         แมท/ธิว/บอก/ว่า/อี/แม่/อย่า/พึ่ง/หลับ/เด้อ   
6    7.0                                       เงียบ/แฮงจัด   
7    8.0          บ่/มี/ความ/รื่น/เริง/อี/หยัง/เลย/ข้าง/นอก   
8    9.0               มื้อ/นี้/สิ/เบิ่ง/พุ/ตอน/หก/ทุ่ม/จ่ะ   
9   10.0             ถ้า/ฝน/ตก/หิ/มะ/ตก/คึ/สิ/บ่/ได้/ออก/ไป   
10  11.0  ปี/ที่/แล้ว/ออก/ไป/เบิ่ง/เขา/จุด/พุ/มัน/เป็น/ฝ...   
11  12.0                        เพิ่น/ก้ม/แบบ/คน/เมา/นะ/จ้ะ   
12  13.0  ก้ม/ก้ม/ส่อง/พุ/เพิ่น/จุด/ละ/เพิ่น/กะ/ก้ม/ส่อง...   
13  14.0            เพิ่น/สิ/โซ/กล้าม/เพิ่น/หั่น/นะ/พี่น้อง   
14  15.0  โอ/เค/จ้า/สุผู้สุคน/มื้อ/นี้/จั่ง/ใด๋/กะ/ขอบ/ค...   
15  16.0  จ้า/พ้อ/กัน/ใหม่/ใน/คลิป/วิ/ดี/โอ/หน้า/สวัสดี/...   

                                              ภาษาไทย     เริ่ม        จบ  \
0                               สวัสดี/ทุก ๆ คน/นะ/คะ  00:00:16  00:00:18   
1           กิน/ข้าว/ก่อน/นะ/จ้ะ/ใคร/จะ/รอ/เค้า/ดาวน์  00:02:14  00:02:18   
2   เค้า/ดาวน์/ก็/ขอ/ให้/เค้า/ดาน์/อย่าง/มี/ความ/ส...  00:02:21  00:02:25   
3   นี่/จ้า/อา/หาร/เย็น/เป็น/ลา/ซาน/ย่า/กับ/ผัก/สด...  00:02:33  00:02:37   
4                         ขอ/ให้/เชิญ/อร่อย/นะ/ทุก/คน  00:02:38  00:02:41   
5           แมท/ธิว/บอก/ว่า/คุณ/แม่/อย่า/พึ่ง/หลับ/นะ  00:03:08  00:03:08   
6                                         เงียบ/มาก ๆ  00:09:51  00:09:53   
7            ไม่/มี/ความ/รื่น/เริง/อะ/ไร/เลย/ด้าน/นอก  00:10:07  00:10:11   
8                  วัน/นี้/จะ/ดู/พุ/ตอน/หก/ทุ่ม/นะ/คะ  00:10:29  00:10:33   
9             ถ้า/ฝน/ตก/หิ/มะ/ตก/คง/จะ/ไม่/ได้/ออก/ไป  00:10:40  00:10:43   
10  ปี/ที่/แล้ว/ออก/ไป/ดู/เขา/จุด/พุ/มัน/เป็น/ฝน/ห...  00:10:52  00:10:56   
11                         เขา/ก้ม/แบบ/คน/เมา/อ่ะ/ค่ะ  00:11:09  00:11:12   
12  ก้ม/ก้ม/ส่อง/พุ/เขา/จุด/ละ/เขา/ก็/ก้ม/ส่อง/เขา...  00:11:14  00:11:18   
13              เขา/จะ/โชว์/กล้าม/เขา/นั่น/แหละ/ทุกคน  00:13:11  00:13:13   
14  โอ/เค/จ้า/ทุก ๆ  คน/วัน/นี้/ยัง/ไง/ก็/ขอบ/คุณ/...  00:15:23  00:15:26   
15  จ้า/เจอ/กัน/ใหม่/ใน/คลิป/วิ/ดี/โอ/หน้า/สวัสดี/...  00:15:30  00:15:33   

    จบ-เริ่ม                                                G2P  \
0   00:00:02  s-a-z^-1 w-a-t^-1 d-ii-z^-0 *s-u-z^-1 *ph-uu-z...   
1   00:00:04  k-i-n^-0 *kh-aa-w^-2 *k-@@-n^-1 *d-qq-z^-2 *c-...   
2   00:00:04  kh-a-w^-3 *d-aa-w^-0 *k-a-z^-1 *kh-@@-z^-4 *h-...   
3   00:00:04  n-ii-z^-2 *c-aa-z^-2 *z-aa-z^-0 *h-aa-n^-4 *j-...   
4   00:00:03  kh-@@-z^-4 *h-a-j^-2 *ch-qq-n^-0 *s-xx-p^-2 *d...   
5   00:00:00  m-xx-t^-1 *th-i-w^-0 *b-@@-k^-1 *w-aa-z^-2 *z-...   
6   00:00:02                  ng-iia-p^-2 *h-xx-ng^-0 *c-a-t^-1   
7   00:00:04  b-@@-z^-1 *m-ii-z^-0 *khw-aa-m^-0 *r-vv-n^-2 *...   
8   00:00:04  m-vv-z^-3 *n-ii-z^-3 *s-i-z^-1 *b-q-ng^-1 *ph-...   
9   00:00:03  th-aa-z^-2 *f-o-n^-4 *t-o-k^-1 *h-i-z^-1 *m-a-...   
10  00:00:04  p-ii-z^-0 *th-ii-z^-2 *l-xx-w^-3 *z-@@-k^-1 *p...   
11  00:00:03  ph-qq-n^-2 *k-o-m^-3 *b-xx-p^-1 *kh-o-n^-0 *m-...   
12  00:00:04  k-o-m^-3 *k-o-m^-3 *s-@-ng^-1 *ph-u-z^-3 *ph-q...   
13  00:00:02  ph-qq-n^-2 *s-i-z^-1 *s-oo-z^-0 *kl-aa-m^-2 *p...   
14  00:00:03  z-oo-z^-0 *kh-ee-z^-0 *c-aa-z^-2\n*s-u-z^-1 *p...   
15  00:00:03  c-aa-z^-2 *ph-@@-z^-3 *k-a-n^-0 *m-a-j^-1 *n-a...   

                                        ภาษาอีสานใหม่  \
0                           สวัสดี,สุผู้สุคน,เด้อ,จ้า   
1        กิน,ข้าว,ก่อน,เด้อ,จ้า, ไผ,สิ,ถ่า,เค้า,ดาวน์   
2   เค้า,ดาว,กะ,ขอ,ให้,เค้า,ดาว,อย่าง,มี,ความ,สุข,...   
3   นี่,จ้า,อา,หาร,เย็น,เป็น,ลา,ซาน,ย่า,กับ,ผัก,สด...   
4                      ขอ,ให้,เชิญ,แซ่บ,เด้อ,พี่,น้อง   
5          แมท,ธิว,บอก,ว่า,อี,แม่,อย่า,พึ่ง,หลับ,เด้อ   
6                                        เงียบ,แฮงจัด   
7           บ่,มี,ความ,รื่น,

## ข้อ 3.2

In [189]:
def splitstr(col1):
    co1 = col1.split('/')
    return co1

In [199]:
splitstr(dt.iloc[1,2])

['กิน', 'ข้าว', 'ก่อน', 'นะ', 'จ้ะ', 'ใคร', 'จะ', 'รอ', 'เค้า', 'ดาวน์']

In [201]:
list_11 = []
list_22 = []
for i in range(0,7):
    list_11 += splitstr(dt.iloc[i,1])
    list_22 += splitstr(dt.iloc[i,2])

In [202]:
def google(col1,col2):
    str = f'{col1} แปลว่า {col2}'
    return str

In [203]:
oil = []
for i in range(len(list_11)):
    oil+= [google(list_11[i],list_22[i])]

In [213]:
google(list_11[1],list_11[2])

'สุผู้สุคน แปลว่า เด้อ'

In [262]:
Uoil = set(oil)

In [263]:
Uoil

{' ไผ แปลว่า ใคร',
 'กะ แปลว่า ก็',
 'กับ แปลว่า กับ',
 'กิน แปลว่า กิน',
 'ก่อน แปลว่า ก่อน',
 'ขอ แปลว่า ขอ',
 'ข้าว แปลว่า ข้าว',
 'ความ แปลว่า ความ',
 'จ้ะ แปลว่า จ้ะ',
 'จ้า แปลว่า คะ',
 'จ้า แปลว่า จ้ะ',
 'จ้า แปลว่า จ้า',
 'ซาน แปลว่า ซาน',
 'ดาว แปลว่า ดาน์',
 'ดาว แปลว่า ดาวน์',
 'ดาวน์ แปลว่า ดาวน์',
 'ถ่า แปลว่า รอ',
 'ธิว แปลว่า ธิว',
 'นะ แปลว่า นะ',
 'นี่ แปลว่า นี่',
 'น้อง แปลว่า คน',
 'บอก แปลว่า บอก',
 'ผัก แปลว่า ผัก',
 'พี่ แปลว่า ทุก',
 'พึ่ง แปลว่า พึ่ง',
 'มี แปลว่า มี',
 'ย่า แปลว่า ย่า',
 'ลา แปลว่า ลา',
 'ว่า แปลว่า ว่า',
 'สด แปลว่า สด',
 'สวัสดี แปลว่า สวัสดี',
 'สิ แปลว่า จะ',
 'สุข แปลว่า สุข',
 'สุผู้สุคน แปลว่า ทุก ๆ คน',
 'หลับ แปลว่า หลับ',
 'หาร แปลว่า หาร',
 'อย่า แปลว่า อย่า',
 'อย่าง แปลว่า อย่าง',
 'อา แปลว่า อา',
 'อี แปลว่า คุณ',
 'เค้า แปลว่า เค้า',
 'เงียบ แปลว่า เงียบ',
 'เชิญ แปลว่า เชิญ',
 'เด้อ แปลว่า นะ',
 'เป็น แปลว่า เป็น',
 'เย็น แปลว่า เย็น',
 'แซ่บ แปลว่า อร่อย',
 'แมท แปลว่า แมท',
 'แม่ แปลว่า แม่',
 'แฮงจัด แปลว่า มาก ๆ',
 'ให้ แปล